sonosco/training/experiment.py

```diff
@@ -99,12 +119,78 @@ def add_directory(self, dir_name):
         # create directory
         path_utils.try_create_directory(dir_path)
 
+    def setup_model_trainer(self,
+                            name: str,
+                            model: torch.nn.Module,
+                            loss: Union[Callable[[Any, Any], Any],
+                                        Callable[[torch.Tensor, torch.Tensor, torch.nn.Module], float]],
+                            epochs: int,
+                            train_data_loader: DataLoader,
+                            model_checkpoints: bool = True,
+                            tensorboard: bool = True,
+                            val_data_loader: DataLoader = None,
+                            decoder  = None,
+                            optimizer = torch.optim.Adam,
+                            lr: float = 1e-4,
+                            custom_model_eval: bool = False,
+                            gpu: int = None,
+                            clip_grads: float = None,
+                            metrics: List[Callable[[torch.Tensor, Any], Union[float, torch.Tensor]]] = None,
+                            callbacks: List[AbstractCallback] = None):
+        """
+        Setup a model_trainer object with specified parameters, by default with checkpoint
+        callback and tensorboard callback. Add this model trainer to the modeltrainers dictionary.
+        """
+
+        trainer = ModelTrainer(model=model, loss=loss, epochs=epochs,
+                               train_data_loader=train_data_loader,
+                               val_data_loader=val_data_loader, decoder=decoder,
+                               optimizer=optimizer, lr=lr, custom_model_eval=custom_model_eval,
+                               gpu=gpu, clip_grads=clip_grads,
+                               metrics=metrics, callbacks=callbacks)
+        if model_checkpoints:
+            date_time = datetime.datetime.fromtimestamp(time()).strftime('%Y-%m-%d_%H:%M:%S')
+            model_name = name + '_' + date_time + '_checkpoint.pt'
+            trainer.add_callback(ModelCheckpoint(output_path=self.logs, model_name=model_name))
+
+        if tensorboard:
+            trainer.add_callback(TensorBoardCallback(log_dir=self.logs))
```

Comment:
there is a plots directory in the experiment class

Reply:
but this is just about the tensorboard log file, or? so there are not really plots saved, or?

Reply:
Yes, we have no plots but I would rather have the logs directory for the actual log files of the program. And for these tensorboard logs I would rather have a separate dir to not mix them up.